src/testdrive/asciidoc.py

```diff
@@ -1,104 +1,424 @@
 ### SPDX-License-Identifier: GPL-2.0-or-later
 
-"""Generate asciidoc output"""
+"""Generate asciidoc output from JUnit inputs"""
 
 from argparse import ArgumentParser
+from collections import OrderedDict
+import os
+from shutil import copyfile
 import json
+from uuid import uuid4
+from decimal import Decimal
+from xml.etree import ElementTree as ET
 
-from .cases import summarize
-from .common import open_input
+class Config(dict):
+    """Configuration of asciidoc generation."""
+    def case_path(self, case):
+        """Return a path in the local filesystem to the files for `case`.
+
+        This function only computes the path given configuration variables:
+        it does not guarantee that the path exists or has required files.
+        """
+        try:
+            test_id = case['test_id']
+            suite = self['suites'][case.suite]
+            baseurl = suite['baseurl']
+            if test_id.startswith(baseurl):
+                return os.path.join(
+                    self['repositories'][suite['repository']],
+                    test_id.split(baseurl, maxsplit=1)[1].lstrip('/'),
+                )
+        except KeyError:
+            pass
+        return None
+    def case_path_testspec(self, case):
```

Comment:
Maybe passing this file into something like `black` would be nice. I think the PEP8 coding style advices for one carriage return to separate class methods and 2 to separate classes.

Reply:
I won't do that in this PR, otherwise it we would have one or two files which conform and all the others not conforming. Propose you raise this in the wider forum.